### ISSUE TITLE
Switch associated to a color item is OFF when ON is expected

### ISSUE THREAD

**Issue by lolodomo** [openhab/openhab.android#153]
I have this item:

```
Color LampeTeleviseur "Lampe TV" {hue="1"}
```

And this in my sitemap:

```
        Text label="Eclairage" {
            Switch item=LampeTeleviseur
        }
```

When I start HABdroid and the physical color light is ON, the switch is in a wrong state OFF.
Then consider that the color light is OFF and the switch in HABdroid is OFF. I switch it to ON in HABdroid, the light switch ON as expected but one second after, the switch in HABdroid switch to OFF (while the physical light remains ON).
The same item and sitemap are working well in Classic UI.

openHAB 1.8.1
HABdroid 1.7.0.1
##

**Comment by lolodomo** [comment]
Same error in HABdroid 1.5.1.1

**Comment by hazzeh** [comment]
Hi,
The problem is that the "state value" from openhab is not parsed correctly. The coloritem uses a HSB value "H,S,B" where B is brightness. When brightness is > 0, the switch should be marked as "ON".

I've got a fix for the problem, and will create a pull request: https://github.com/hazzeh/openhab.android/tree/153-Color_items_state
Until it is merged into master and delivered you can build the application from source if you like.

**Comment by lolodomo** [comment]
@hazzeh thank you

Do you know if it exists nightly builds for the Android app ?

**Comment by hazzeh** [comment]
I think you can find "nightly" builds at https://openhab.ci.cloudbees.com/job/HABDroid/ but my commit is not merged into master yet, so the fix is not available.

**Comment by CristianSarac** [comment]
@hazzeh Have you managed to build the App from source? 
I am trying to build it from Android Studio and i always get this error 
`FATAL EXCEPTION: main java.lang.NoClassDefFoundError: org.openhab.habdroid.util.MySSLSocketFactory
at org.openhab.habdroid.util.MyAsyncHttpClient.<init>(MyAsyncHttpClient.java:38)
at org.openhab.habdroid.ui.OpenHABMainActivity.onCreate(OpenHABMainActivity.java:205)
                                                                          at android.app.Activity.performCreate(Activity.java:5141)
                                                                          at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1081)
                                                                          at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2270)
                                                                          at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2358)
                                                                          at android.app.ActivityThread.access$600(ActivityThread.java:156)
                                                                          at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1340)
                                                                          at android.os.Handler.dispatchMessage(Handler.java:99)
                                                                          at android.os.Looper.loop(Looper.java:153)
                                                                          at android.app.ActivityThread.main(ActivityThread.java:5297)
                                                                          at java.lang.reflect.Method.invokeNative(Native Method)
                                                                          at java.lang.reflect.Method.invoke(Method.java:511)
                                                                          at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:833)
                                                                          at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:600)
                                                                          at dalvik.system.NativeStart.main(Native Method)`

**Comment by hazzeh** [comment]
> @hazzeh Have you managed to build the App from source? 

Yes, but I had to do a minor update to the build configuration: https://github.com/hazzeh/openhab.android/commit/9913fbe5d7f774f7b197f5f45a1b23349a38d4d4
Also I'm building within android studio 1.5.1.
SDK with api level 23
Android SDK tools 24.4.1
Android SDK Platform tools 23.1.0

I've asked a question in the form about building from source: https://community.openhab.org/t/problem-building-android-app-from-source/8327

**Comment by kaikreuzer** [comment]
I have no idea myself about the android build, but in general there is a build plan on Cloudbees that is supposed to build it: https://openhab.ci.cloudbees.com/job/HABDroid/
And so far it looks green. Not sure if this helps you, though...

**Comment by teichsta** [comment]
> And so far it looks green. Not sure if this helps you, though...

last successful build is from `Oct 18, 2015 10:51 PM`

**Comment by hazzeh** [comment]
It seems that the build plan for Habdroid on Cloudbees is only for the master branch, and there are no build for pull requests.
But when a PR is merged into master, the apk can be fetched from cloudbees prior to it reaches google play store.

**Comment by kaikreuzer** [comment]
> last successful build is from Oct 18, 2015 10:51 PM

Well, this was the last code change on the app - which is indeed a shame :cry:, so there is nothing wrong with the build plan in general.

**Comment by teichsta** [comment]
> so there is nothing wrong with the build plan in general.

ah yes … indeed!

**Comment by lolodomo** [comment]
@hazzeh : unforutnatelly my issue is still not fixed in version 1.8.

I don't see how to re-open this issue.

**Comment by digitaldan** [comment]
@lolodomo, it's open again.

**Comment by hazzeh** [comment]
For me, it works with openhab 2 and a habdroid I've built from the gitrepo (should be the same as one of the nightly builds.
Unfortunately I don't have physical access to my smart home the next 2 weeks.
When I get physical access to my smarthome, I can check with the official 1.8 version of habdroid, and openhab 1.x. And see if I notice anything.

**Comment by lolodomo** [comment]
@hazzeh : I can confirm that the fix works well in openHAB 2 beta 3. But unfortunately it does not work in openHAB 1.8.3.
Does it make sense that the behaviour is different with 1.8 server and 2.0 server ?

**Comment by hazzeh** [comment]
@lolodomo I'm almost sure it is not bug in habdroid. When I tried 1.8.3 and a hue binding, the SwitchItem is not updated (checking the rest api). When I tried with 1.8.0 everything works.
Maybe it should be an open issue for openhab 1.8.3 (I've not tried any other version).

**Comment by hazzeh** [comment]
@lolodomo I did some testing, and it seems to be some glitches with the hue binding for openhab 1.8.3 (it is fixed in 1.9.0-snapshot). Did you do a fresh install of 1.8.3 and paired the hue bridge? If you hue bridge is using the latest firmware the 1.8.3 binding won't work, see: https://github.com/openhab/openhab/pull/4311#issuecomment-222287612
What I did to get 1.8.3 working, was to use the 1.9.0 hue binding when doing the initial pairing (with a new secret, then the log told me to change he secret, after that the hue statuses was updated correctly (both in webui and habdroid).
After that I tested to switch back to hue binding 1.8.3 which also worked.
You can try that.
@digitaldan I think we can close this issue again.

**Comment by lolodomo** [comment]
Time to close this issue.
OH 1.8.3 os now obsolete.
No problem with OH 2.
